src/FixedPointDecimals.jl

```diff
@@ -441,6 +441,29 @@ Base.Checked.mul_with_overflow(x::FD, y::FD) = Base.Checked.mul_with_overflow(pr
 Base.Checked.mul_with_overflow(x::FD, y) = Base.Checked.mul_with_overflow(promote(x, y)...)
 Base.Checked.mul_with_overflow(x, y::FD) = Base.Checked.mul_with_overflow(promote(x, y)...)
 
+# This does not exist in Base so is just part of this package.
+# Throws on divide by zero.
+@doc """
+    div_with_overflow(x::FixedDecimal{T,f}, y::FixedDecimal{T,f})::(FixedDecimal{T,f}, Bool)
+        where {T<:Integer, f}
+
+Return the result of div (wrapping on overflow/underflow) and a boolean indicating whether
+overflow/underflow did in fact happen. Throws a DivideError on divide-by-zero.
+"""
+function div_with_overflow(x::FD{T,f}, y::FD{T,f}) where {T<:Integer,f}
+    C = coefficient(FD{T, f})
+    # This case will break the div call below.
+    if x.i == typemin(T) && y.i == -1
+        # To perform the div and overflow means reaching the max and adding 1, so typemin.
+        return (typemin(FD{T,f}), true)
```

Comment:
```suggestion
        return (x, true)
```
i think this would be the same code anyway, but it makes it a bit simpler to read